hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncTableBuilder.java

```diff
@@ -137,6 +138,11 @@ default AsyncTableBuilder<C> setMaxRetries(int maxRetries) {
    */
   AsyncTableBuilder<C> setStartLogErrorsCnt(int startLogErrorsCnt);
 
+  /**
+   * Sets the map of request attributes
+   */
+  AsyncTableBuilder<C> setRequestAttributes(Map<String, byte[]> requestAttributes);
```

Comment:
What do you think about `setRequestAttribute(String, byte[])` so that one can more fluently create their Table/AsyncTable if desired? One can chain together ImmutableMap.builder().add().add().build, but not everyone might have it and it's a bit more verbose.

Reply:
We prob could leave all the *Caller classes as you have them since they are internal and post-build, but for the TableBuilder/AsyncTableBuilder it might be nice.

Reply:
Yeah I'd be happy to add that. Just to clarify, we should probably support both, right? And calling setRA after addRA would implicitly overwrite the individual additions?

Reply:
I don't have a strong opinion on whether we should continue to support `setRequestAttributes` (with an s). Do you think that has a particular value as is?

If we do support both, then yea we need to figure out the interplay. I agree that calling set after add would overwrite, but what about add after set? One could imagine someone calling set with an immutable map, so we'd probably have to copy it in order to ensure add operations work.

Reply:
>  One could imagine someone calling set with an immutable map, so we'd probably have to copy it in order to ensure add operations work.

Yeah that's a good point. 

> I don't have a strong opinion on whether we should continue to support setRequestAttributes (with an s). Do you think that has a particular value as is?

I don't feel _particularly_ strongly, but it seems plausible that someone will have a map of requests pre-constructed, particularly if they have several attributes to pass, and that it would be nicer to pass it in than need to iterate calls to addRA

Reply:
If we provide a `setRequestAttribute(String, byte[])` method then do we also want to support unsetting a key?

Reply:
Just pushed a change which allows for the two methods `setRequestAttribute` and `setRequestAttributes`. The plural will implicitly overwrite anything already submitted. It also enforces the implementation of map that we're working with

Reply:
Also, I think you need to update the ThriftConnection to override the new method.